homeassistant/components/home_plus_control/switch.py

```diff
@@ -51,56 +46,43 @@ async def async_update_data():
             # Note: asyncio.TimeoutError and aiohttp.ClientError are already
             # handled by the data update coordinator.
             async with async_timeout.timeout(10):
-                switch_data = await api.fetch_data()
-        except HomeAssistantError as err:
+                module_data = await api.fetch_data()
+        except HomePlusControlApiError as err:
             raise UpdateFailed(
                 f"Error communicating with API: {err} [{type(err)}]"
             ) from err
 
+        current_entity_uids = set(hass.data[DOMAIN][config_entry.entry_id][ENTITY_UIDS])
```

Comment:
This check should be done in the init module and then we should signal the platforms with our dispatch helper if entities need to be created.

Removing devices and entities will be done via the device registry so that can be called directly without a signal I think.

Reply:
The coordinator update is now in the init module and the platform signals for the addition of new entities. 
I removed the signalling of the removal of devices and it is now directly performed in the coordinator update method.